### PR TITLE
Visualization spec review

### PR DESCRIPTION
**Pull request by javitonino**
Reenabled visualization_spec tests, as we discovered we didn't have proper tests for visualization deletion. 

Fixed a bug with default privacy for visualizations: it should be the privacy of the table by default, and only use the default privacy as a fallback.

### CODE REVIEW

**Comment by javitonino** [comment]
Please CR @juanignaciosl

File: app/models/visualization/member.rb [hunk 1]
```diff
@@ -160,6 +160,11 @@ def ==(other_vis)
         self.id == other_vis.id
       end
 
+      def ensure_valid_privacy(owner)
```
Comment (juanignaciosl) [inline app/models/visualization/member.rb]: Shouldn't this be based on `self.owner`?

**Comment by javitonino** [comment]
Changed `ensure_valid_privacy` so it uses the visualization owner, thanks! Second look @juanignaciosl?

File: app/controllers/api/json/visualizations_controller.rb [hunk 2]
```diff
@@ -71,7 +71,7 @@ def create
                 Visualization::Member.new(vis_data.merge(name: name_candidate, user_id:  current_user_id))
               end
 
-        vis.privacy = vis.default_privacy(current_user)
+        vis.ensure_valid_privacy
```
Comment (juanignaciosl) [inline app/controllers/api/json/visualizations_controller.rb]: Sorry about not seeing this before, but when the parameter is gone it becomes more evident. This obviously belongs to model creation. Is there any technical impediment for it?

Comment (javitonino) [inline app/controllers/api/json/visualizations_controller.rb]: The privacy and user permissions are already validated at the model. I wouldn't go so far as auto-correcting the privacy in the validation, better to throw an error, so I think calling this from here makes sense.

I can put the default_privacy part into the visualization `initialize`, but I'd keep the `ensure_valid_privacy` call here. Auto-correcting privacy in the model seems wrong, better to throw a validation error.

**Pull request merged**